### PR TITLE
Fix: Move `setup.cfg` to `pyproject.toml`

### LINKED ISSUE #97: Move configuration from `setup.cfg` to `pyproject.toml`

### PR DESCRIPTION
**Pull request by sumit-158**
Description:
- Fix #97

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -46,3 +46,26 @@ path = "broadcaster/__init__.py"
...
+module = "tests.*"
+disallow_untyped_defs = false
+check_untyped_defs = true
+
+[tool.pytest.ini_options]
+addopts = "-rxXs"
+markers = "copied_from(source, changes=None): mark test as copied from somewhere else, along with a description of changes made to accodomate e.g. our test setup"
+
+[tool.coverage.run]
+source_pkgs = ["broadcaster", "tests"]
```
Comment (Kludex) [inline pyproject.toml:71]:
```suggestion
source_pkgs = ["broadcaster", "tests"]

```

**Review by Kludex — CHANGES_REQUESTED** [review]
Can you remove the `setup.cfg`? 🙏

**Comment by sumit-158** [comment]
> Can you remove the `setup.cfg`? 🙏

Oops :grin: I missed that. Thank you!

**Comment by Kludex** [comment]
I don't think ruff is being used in the pipeline yet.

It still needs to remove flake8 and isort from the project (requirements.txt and somewhere in scripts), and add ruff.

**Comment by sumit-158** [comment]
@Kludex Why does the coverage decrease?

**Comment by sumit-158** [comment]
Should I decrease `--fail-under=88` under 78?

File: scripts/lint [hunk 2]
```diff
@@ -8,6 +8,5 @@ export SOURCE_FILES="broadcaster tests"
 
 set -x
 
-${PREFIX}autoflake --in-place --recursive $SOURCE_FILES
-${PREFIX}isort --project=broadcaster $SOURCE_FILES
-${PREFIX}black --target-version=py37 $SOURCE_FILES
+${PREFIX}ruff --fix $SOURCE_FILES
+${PREFIX}black --target-version=py37 $SOURCE_FILES
```
Comment (Kludex) [inline scripts/lint:12]:
```suggestion
${PREFIX}black --target-version=py37 $SOURCE_FILES

```

File: scripts/check [hunk 3]
```diff
@@ -9,6 +9,5 @@ export SOURCE_FILES="broadcaster tests"
 set -x
 
 ${PREFIX}black --check --diff --target-version=py37 $SOURCE_FILES
-${PREFIX}flake8 $SOURCE_FILES
-${PREFIX}mypy $SOURCE_FILES
-${PREFIX}isort --check --diff --project=broadcaster $SOURCE_FILES
+${PREFIX}ruff check $SOURCE_FILES
+${PREFIX}mypy $SOURCE_FILES
```
Comment (Kludex) [inline scripts/check:13]:
```suggestion
${PREFIX}mypy $SOURCE_FILES

```

File: requirements.txt [hunk 4]
```diff
@@ -10,13 +10,9 @@ build==0.10.0
 twine==4.0.1
 
 # Tests & Linting
-autoflake==1.4
+ruff==0.0.272
```
Comment (Kludex) [inline requirements.txt:13]:
```suggestion
ruff==0.0.277
```

**Review by Kludex — CHANGES_REQUESTED** [review]
Yep, you can change the coverage. I'll open an issue to increase the coverage to 100% after this is merged.

**Comment by Kludex** [comment]
Please rebase.

File: scripts/coverage [hunk 5]
```diff
@@ -8,4 +8,4 @@ export SOURCE_FILES="broadcaster tests"
 
 set -x
 
-${PREFIX}coverage report --show-missing --skip-covered --fail-under=88
+${PREFIX}coverage report --show-missing --skip-covered --fail-under=78
```
Comment (Kludex) [inline scripts/coverage:11]: Actually, can you move `show-missing`, `skip-covered` and `fail-under` to the `pyproject.toml`?

File: scripts/coverage [hunk 6]
```diff
@@ -8,4 +8,4 @@ export SOURCE_FILES="broadcaster tests"
 
 set -x
 
-${PREFIX}coverage report --show-missing --skip-covered --fail-under=78
+${PREFIX}coverage report
```
Comment (Kludex) [inline scripts/coverage:11]: Can we also remove the `SOURCE_FILES` from this file, please?

Just noticed, sorry.

Comment (sumit-158) [reply scripts/coverage:10]: No worries :-)

**Review by Kludex — APPROVED** [review]

**Comment by Kludex** [comment]
Thanks @sumit-158 🙏

**Pull request merged**